services/hanke-service/Dockerfile-local

```diff
@@ -13,7 +13,7 @@ WORKDIR /builder
 # Add GRADLE_OPTS to prevent JVM forking while building
 RUN GRADLE_OPTS="-XX:MaxMetaspaceSize=256m -XX:+HeapDumpOnOutOfMemoryError -Xms256m -Xmx512m" ./gradlew assemble --no-daemon
 
-FROM openjdk:11
+FROM eclipse-temurin:17
```

Comment:
```
RUN ./gradlew --version
RUN java -version
```

Lines 11-12. Are these trying to log the versions? I don't think they work as expected. Logs: 

```
 => [haitaton-hanke builder 4/8] WORKDIR /builder                                                                                                                                                                                        0.0s
 => [haitaton-hanke builder 5/8] RUN ./gradlew --version     
```

Maybe now is a good time to remove them.

Reply:
The default TTY progress setting hides the output, which doesn't mean they don't work. Setting the progress to plain shows the output.

Their real purpose is to provide compose with cacheable checkpoints, which it uses at varying success rates. E.g. the Gradle binary is downloaded at the version check step.
```
❯ docker-compose build haitaton-hanke --progress plain
#1 [internal] load .dockerignore
#1 transferring context: 2B done
#1 DONE 0.0s

#2 [internal] load build definition from Dockerfile-local
#2 transferring dockerfile: 739B done
#2 DONE 0.1s

#3 [internal] load metadata for docker.io/library/eclipse-temurin:17
#3 DONE 2.6s

#4 [builder 1/9] FROM docker.io/library/eclipse-temurin:17@sha256:2afd828b3c10d72c0dc2d0c297bb2e6633f9f22609da1202dddd94dd78d4627d
#4 DONE 0.0s

#5 [internal] load build context
#5 transferring context: 101.90MB 2.3s
#5 transferring context: 108.76MB 2.4s done
#5 DONE 2.4s

#4 [builder 1/9] FROM docker.io/library/eclipse-temurin:17@sha256:2afd828b3c10d72c0dc2d0c297bb2e6633f9f22609da1202dddd94dd78d4627d
#4 CACHED

#6 [builder 2/9] COPY *.gradle gradle.* gradlew /builder/
#6 DONE 1.1s

#7 [builder 3/9] COPY gradle /builder/gradle
#7 DONE 0.0s

#8 [builder 4/9] WORKDIR /builder
#8 DONE 0.0s

#9 [builder 5/9] RUN ./gradlew --version
#9 0.323 Downloading https://services.gradle.org/distributions/gradle-8.1.1-bin.zip
#9 0.857 ...........10%............20%............30%............40%............50%............60%............70%............80%...........90%............100%
#9 15.57
#9 15.57 Welcome to Gradle 8.1.1!
#9 15.57
#9 15.57 Here are the highlights of this release:
#9 15.57  - Stable configuration cache
#9 15.57  - Experimental Kotlin DSL assignment syntax
#9 15.57  - Building with Java 20
#9 15.57
#9 15.57 For more details see https://docs.gradle.org/8.1.1/release-notes.html
#9 15.57
#9 15.57
#9 15.57 ------------------------------------------------------------
#9 15.57 Gradle 8.1.1
#9 15.57 ------------------------------------------------------------
#9 15.57
#9 15.57 Build time:   2023-04-21 12:31:26 UTC
#9 15.57 Revision:     1cf537a851c635c364a4214885f8b9798051175b
#9 15.57
#9 15.57 Kotlin:       1.8.10
#9 15.57 Groovy:       3.0.15
#9 15.57 Ant:          Apache Ant(TM) version 1.10.11 compiled on July 10 2021
#9 15.57 JVM:          17.0.7 (Eclipse Adoptium 17.0.7+7)
#9 15.57 OS:           Linux 5.15.49-linuxkit aarch64
#9 15.57
#9 DONE 15.7s

#10 [builder 6/9] RUN java -version
#10 0.328 openjdk version "17.0.7" 2023-04-18
#10 0.328 OpenJDK Runtime Environment Temurin-17.0.7+7 (build 17.0.7+7)
#10 0.328 OpenJDK 64-Bit Server VM Temurin-17.0.7+7 (build 17.0.7+7, mixed mode, sharing)
#10 DONE 0.3s

#11 [builder 7/9] ADD . /builder
#11 DONE 24.8s

#12 [builder 8/9] WORKDIR /builder
#12 DONE 0.0s

#13 [builder 9/9] RUN GRADLE_OPTS="-XX:MaxMetaspaceSize=256m -XX:+HeapDumpOnOutOfMemoryError -Xms256m -Xmx512m" ./gradlew assemble --no-daemon
#13 0.711 To honour the JVM settings for this build a single-use Daemon process will be forked. See https://docs.gradle.org/8.1.1/userguide/gradle_daemon.html#sec:disabling_the_daemon.
#13 1.210 Daemon will be stopped at the end of the build
#13 51.02 > Task :services:hanke-service:bootBuildInfo
#13 71.31 > Task :services:hanke-service:processResources
#13 81.11 > Task :services:hanke-service:compileKotlin
#13 86.81 > Task :services:hanke-service:compileJava NO-SOURCE
#13 86.81 > Task :services:hanke-service:classes
#13 88.61 > Task :services:hanke-service:bootJarMainClassName
#13 89.41 > Task :services:hanke-service:bootJar
#13 89.41 > Task :services:hanke-service:jar
#13 89.41 > Task :services:hanke-service:assemble
#13 89.50
#13 89.50 BUILD SUCCESSFUL in 1m 29s
#13 89.50 6 actionable tasks: 6 executed
#13 DONE 90.6s

#14 [stage-1 2/4] WORKDIR /app
#14 CACHED

#15 [stage-1 3/4] COPY --from=builder /builder/services/hanke-service/build/libs/hanke-service-*.jar /app/haitaton.jar
#15 DONE 0.2s

#16 [stage-1 4/4] COPY --from=builder /builder/scripts/wait-for-it.sh /app/wait-for-it.sh
#16 DONE 0.0s

#17 exporting to image
#17 exporting layers
#17 exporting layers 0.1s done
#17 writing image sha256:02ac360243b111e0dedc95feda95e1d1fde83b3b02ba65fd2b7031bc30399cad done
#17 naming to docker.io/services/hanke-service done
#17 DONE 0.2s
```

At next build:
```
❯ docker-compose build haitaton-hanke --progress plain
#1 [internal] load .dockerignore
#1 transferring context: 2B done
#1 DONE 0.0s

#2 [internal] load build definition from Dockerfile-local
#2 transferring dockerfile: 739B done
#2 DONE 0.0s

#3 [internal] load metadata for docker.io/library/eclipse-temurin:17
#3 DONE 2.2s

#4 [builder 1/9] FROM docker.io/library/eclipse-temurin:17@sha256:2afd828b3c10d72c0dc2d0c297bb2e6633f9f22609da1202dddd94dd78d4627d
#4 DONE 0.0s

#5 [internal] load build context
#5 transferring context: 720.36kB 0.4s done
#5 DONE 0.4s

#6 [builder 2/9] COPY *.gradle gradle.* gradlew /builder/
#6 CACHED

#7 [builder 3/9] COPY gradle /builder/gradle
#7 CACHED

#8 [builder 4/9] WORKDIR /builder
#8 CACHED

#9 [builder 5/9] RUN ./gradlew --version
#9 CACHED

#10 [builder 6/9] RUN java -version
#10 CACHED

#11 [builder 7/9] ADD . /builder
#11 DONE 16.9s

#12 [builder 8/9] WORKDIR /builder
#12 DONE 0.0s

#13 [builder 9/9] RUN GRADLE_OPTS="-XX:MaxMetaspaceSize=256m -XX:+HeapDumpOnOutOfMemoryError -Xms256m -Xmx512m" ./gradlew assemble --no-daemon
#13 0.771 To honour the JVM settings for this build a single-use Daemon process will be forked. See https://docs.gradle.org/8.1.1/userguide/gradle_daemon.html#sec:disabling_the_daemon.
#13 1.471 Daemon will be stopped at the end of the build
#13 53.78 > Task :services:hanke-service:bootBuildInfo
#13 72.87 > Task :services:hanke-service:processResources
#13 83.77 > Task :services:hanke-service:compileKotlin
#13 88.37 > Task :services:hanke-service:compileJava NO-SOURCE
#13 88.37 > Task :services:hanke-service:classes
#13 90.17 > Task :services:hanke-service:bootJarMainClassName
#13 90.77 > Task :services:hanke-service:bootJar
#13 90.87 > Task :services:hanke-service:jar
#13 90.87 > Task :services:hanke-service:assemble
#13 90.87
#13 90.87 BUILD SUCCESSFUL in 1m 30s
#13 90.87 6 actionable tasks: 6 executed
#13 DONE 91.9s

#14 [stage-1 2/4] WORKDIR /app
#14 CACHED

#15 [stage-1 3/4] COPY --from=builder /builder/services/hanke-service/build/libs/hanke-service-*.jar /app/haitaton.jar
#15 DONE 0.2s

#16 [stage-1 4/4] COPY --from=builder /builder/scripts/wait-for-it.sh /app/wait-for-it.sh
#16 DONE 0.0s

#17 exporting to image
#17 exporting layers 0.1s done
#17 writing image sha256:34df618b1e03ff38cc185d08f5c3e88b11ec2a8d15d9d8d3a477611ed8b7ea71 done
#17 naming to docker.io/services/hanke-service done
#17 DONE 0.1s
```